x-pack/legacy/plugins/ml/public/data_visualizer/page.tsx

```diff
@@ -0,0 +1,672 @@
+          // Document count card.
+          configWithStats.stats = metricFieldStats.find(
+            (fieldStats: any) => fieldStats.fieldName === undefined
+          );
+
+          // Add earliest / latest of timefilter for setting x axis domain.
+          configWithStats.stats.timeRangeEarliest = earliest;
+          configWithStats.stats.timeRangeLatest = latest;
+        }
+        configWithStats.loading = false;
+        configs.push(configWithStats);
+      });
+
+      setMetricConfigs(configs);
+    } catch (err) {
+      dataLoader.displayError(err);
+    }
+  }
+
+  async function loadNonMetricFieldStats() {
+    // Only request data for fields that exist in documents.
+    if (nonMetricConfigs.length === 0) {
+      return;
+    }
+
+    const configsToLoad = nonMetricConfigs.filter(
+      config => config.existsInDocs === true && config.loading === true
+    );
+    if (configsToLoad.length === 0) {
+      return;
+    }
+
+    // Pass the field name, type and cardinality in the request.
+    // Top values will be obtained on a sample if cardinality > 100000.
+    const existNonMetricFields: FieldRequestConfig[] = configsToLoad.map(config => {
+      const props = { fieldName: config.fieldName, type: config.type, cardinality: 0 };
+      if (config.stats !== undefined && config.stats.cardinality !== undefined) {
+        props.cardinality = config.stats.cardinality;
+      }
+      return props;
+    });
+
+    const tf = timefilter as any;
+    let earliest;
+    let latest;
+    if (currentIndexPattern.timeFieldName !== undefined) {
+      earliest = tf.getActiveBounds().min.valueOf();
+      latest = tf.getActiveBounds().max.valueOf();
+    }
+
+    try {
+      const nonMetricFieldStats = await dataLoader.loadFieldStats(
+        searchQuery,
+        samplerShardSize,
+        earliest,
+        latest,
+        existNonMetricFields
+      );
+
+      // Add the field stats to the existing stats in the corresponding config.
+      const configs: FieldVisConfig[] = [];
+      nonMetricConfigs.forEach(config => {
+        const configWithStats = { ...config };
+        if (config.fieldName !== undefined) {
+          configWithStats.stats = {
+            ...configWithStats.stats,
+            ...nonMetricFieldStats.find(
+              (fieldStats: any) => fieldStats.fieldName === config.fieldName
+            ),
+          };
+        }
+        configWithStats.loading = false;
+        configs.push(configWithStats);
+      });
+
+      setNonMetricConfigs(configs);
+    } catch (err) {
+      dataLoader.displayError(err);
+    }
+  }
+
+  function createMetricCards() {
+    const configs: FieldVisConfig[] = [];
+    const aggregatableExistsFields: any[] = overallStats.aggregatableExistsFields || [];
+
+    let allMetricFields = indexPatternFields.filter(f => {
+      return (
+        f.type === KBN_FIELD_TYPES.NUMBER &&
+        (f.displayName !== undefined && !DataLoader.OMIT_FIELDS.includes(f.displayName))
```

Comment:
rather than making `OMIT_FIELDS` public and repeating this `includes` call, a method could be added to `DataLoader` to do this check.
e.g. `dataLoader.isAllowedField(f.displayName)`

Reply:
Done in https://github.com/elastic/kibana/pull/42685/commits/992c67e94e727b7b5b5f91cbc73eb954c6e13a0f